tensorboard/plugins/interactive_inference/witwidget/notebook/jupyter/js/lib/wit.js

```diff
@@ -0,0 +1,163 @@
+          this.touch();
+        });
+        this.view_.addEventListener('get-eligible-features', e => {
+          let i = this.model.get('get_eligible_features') + 1;
+          this.model.set('get_eligible_features', i);
+          this.touch();
+        });
+        this.view_.addEventListener('infer-mutants', e => {
+          e.detail['infer_mutants_counter'] = this.inferMutantsCounter++;
+          this.model.set('infer_mutants', e.detail);
+          this.mutantFeature = e.detail.feature_name;
+          this.touch();
+        });
+        setTimeout(()=> {
+          this.config_changed();
+          this.examples_changed();
+          this.sprite_changed();
+        }, 0);
+    },
+
+    // Callback functions for when changes made on python side.
+    examples_changed: function() {
+      const examples = this.model.get('examples');
+      if (!this.view_.updateExampleContents_) {
+        setTimeout(() => this.examples_changed(), 100);
+        return;
+      }
+      if (examples && examples.length > 0) {
+        this.view_.updateExampleContents_(examples, false);
+      }
+    },
+    inferences_changed: function() {
+      const inferences = this.model.get('inferences');
+      this.view_.labelVocab = inferences['label_vocab'];
+      this.view_.inferences = inferences['inferences'];
+    },
+    eligible_features_changed: function() {
+      const features = this.model.get('eligible_features');
+      this.view_.partialDepPlotEligibleFeatures = features
+    },
+    mutant_charts_changed: function() {
+      const chartInfo = this.model.get('mutant_charts');
+      this.view_.makeChartForFeature_(chartInfo.chartType, this.mutantFeature,
+          chartInfo.data);
+    },
+    config_changed: function() {
+      const config = this.model.get('config');
+      if ('inference_address' in config) {
+        let addresses = config['inference_address'];
+        if ('inference_address_2' in config) {
+          addresses += ',' + config['inference_address_2'];
+        }
+        this.view_.inferenceAddress = addresses;
+      }
+      if ('model_name' in config) {
+        let names = config['model_name'];
+        if ('model_name_2' in config) {
+          names += ',' + config['model_name_2'];
+        }
+        this.view_.modelName = names;
+      }
+      if ('model_type' in config) {
+        this.view_.modelType = config['model_type'];
+      }
+      if ('are_sequence_examples' in config) {
+        this.view_.sequenceExamples = config['are_sequence_examples'];
+      }
+      if ('max_classes' in config) {
+        this.view_.maxInferenceEntriesPerRun = config['max_classes'];
+      }
+      if ('multiclass' in config) {
+        this.view_.multiClass = config['multiclass'];
+      }
+      this.view_.updateNumberOfModels_();
+    },
+    sprite_changed: function() {
+      if (!this.view_.updateSprite_) {
+        setTimeout(() => this.sprite_changed(), 100);
```

Comment:
consider clearing timeout before setting a new one.

Reply:
switched to requestAnimationFrame